packages/number-field/src/vaadin-number-field-mixin.js

```diff
@@ -106,12 +112,13 @@ export const NumberFieldMixin = (superClass) =>
     }
 
     /**
-     * Whether the input element's value is unparsable.
+     * The input element's value when it cannot be parsed as a number, and an empty string otherwise.
      *
+     * @return {string}
      * @private
      */
-    get __hasUnparsableValue() {
-      return this.inputElement.validity.badInput;
+    get __unparsableValue() {
```

Comment:
I found the previous approach with the boolean getter a bit clearer, and it looks like we don't necessarily have to change this. Having an unparsable value string if the value is parsable is potentially confusing. A boolean expresses this clearer because either the value parsable or not, and that is all we are interested in.

Reply:
Thanks, reverted the previous approach and also added an undefined check to the `_inputElementValue` getter that's necessary for Flow.